File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/wallet/WalletConfig.java
package com.evernym.verity.sdk.wallet;

import org.json.JSONObject;

/**
 * The generic interface for the wallet configuration for the VDR wallet
 */
public interface WalletConfig {
    /**
     * Provides a JSON string that is valid config for the VDR wallet API
     *
     * @return a JSON wallet config string
     */
    String config();

    /**
     * Provides a JSON string that is valid credential for the VDR wallet API
     *
     * @return a JSON wallet credential string
     */
    String credential();

    /**
     * Add relevant fields to a JSON object to express this configuration
     * @param json the JSON object that the fields are injected into
     */
    void addToJson(JSONObject json);
}

File: sdk/dotnet-sdk/src/VeritySDK/Wallet/WalletConfig.cs
using System;
using System.Json;

namespace VeritySDK
{
    /// <summary>
    /// The generic interface for the wallet configuration for the Indy-SDK wallet
    /// </summary>
    public interface WalletConfig
    {
        /// <summary>
        /// Provides a JSON string that is valid config for the Indy-SDK wallet API
        /// </summary>
        /// <returns>JSON wallet config string</returns>
        String config();

        /// <summary>
        /// Provides a JSON string that is valid credential for the Indy-SDK wallet API
        /// </summary>
        /// <returns>JSON wallet credential string</returns>
        String credential();

        /// <summary>
        /// Add relevant fields to a JSON object to express this configuration
        /// </summary>
        /// <returns>json the JSON object that the fields are injected into</returns>
        void addToJson(JsonObject json);
    }
}
